File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem39_BinaryTreeRightSideViewTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem39_BinaryTreeRightSideViewTest {

    private final Problem39_BinaryTreeRightSideView test = new Problem39_BinaryTreeRightSideView();

    @Test
    void rightSideView() {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(1);
        root.left.right = new TreeNode(5);
        root.right = new TreeNode(3);
        root.right.right = new TreeNode(4);

        List<Integer> list = Arrays.asList(1, 3, 4);

        assertEquals(list, test.rightSideView(root));
    }
}

File: JavaScript/leetcode75/src/test/Problem39_BinaryTreeRightSideView.test.js
const rightSideView = require("../main/Problem39_BinaryTreeRightSideView");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return the right side view values", () => {
  let root = new TreeNode(1);
  root.left = new TreeNode(2);
  root.left.right = new TreeNode(5);
  root.right = new TreeNode(3);
  root.right.right = new TreeNode(4);

  expect(rightSideView(root)).toStrictEqual([1, 3, 4]);
});
